sway-ir/src/optimize/simplify_cfg.rs

```diff
@@ -220,18 +220,14 @@ fn merge_blocks(context: &mut Context, function: &Function) -> Result<bool, IrEr
                 instr.parent = from_block;
             }
 
-            // Re-get the block contents mutably.
-            let (from_contents, to_contents) = context.blocks.get2_mut(from_block.0, to_block.0);
-            let from_contents = from_contents.unwrap();
-            let to_contents = to_contents.unwrap();
-
             // Drop the terminator from `from_block`.
-            from_contents.instructions.pop();
+            from_block.remove_instruction_at(context, from_block.num_instructions(context) - 1);
```

Comment:
This pattern occurs several times. Perhaps having a dedicated method `remove_last_instruction`?